joblib/test/test_memory.py

```diff
@@ -162,14 +162,12 @@ def name_collision(x):
 
     b = name_collision
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
+    with warns(JobLibCollisionWarning) as warninfo:
         a(1)
         b(1)
 
-        assert len(w) == 1
-        assert "collision" in str(w[-1].message)
+    assert len(warninfo) == 1
+    assert "collision" in str(warninfo[-1].message)
```

Comment:
Use `warninfo[0]`. It's the same since there is only one warning but it's less surprising.